plugins/rplanners/rrt.h

```diff
@@ -456,6 +457,13 @@ Some python code to display data::\n\
             // extend A
             ExtendType et = TreeA->Extend(_sampleConfig, iConnectedA);
 
+            if (et == ET_Failed && IS_DEBUGLEVEL(Level_Verbose)) {
+                std::vector<dReal> robotJointValues;
+                this->_getstatefn(robotJointValues); // Making the assumption that _getstatefn is shared throughout planning, e.g. shares same _getstatefn as DynamicsCollisionConstraint checker
+                planningstatus.vRobotJointValues.push_back(robotJointValues); // Is this slow?
+                planningstatus.vCollisionReports.push_back(this->_treeForward.GetCollisionReportString());
+            }
+
```

Comment:
Why should we collect joint values every time a tree extension fails?

Reply:
I was thinking that we could visualize the locations where collision happens on the UI (though according to Kei and Antoine, teachworker team has a tool in which they dump the tree and visualize all the nodes)

Reply:
The joint values collected here are ones that are _not_ in the tree. I think it will rarely be useful/informative to look at these joint values.

Tree dumping functionality is already there and can be used by `planner->SendCommand("DumpTree")`.

What problems are you trying to solve that require these information/ will be easier to solve with these information?

Reply:
I see...I was hoping to collect the associated joint values for every sampled configuration, so that later we can move the robot to those joint locations and see where the robot (or transferred workpiece ) is colliding during planning (and essentially visualize where in the configuration space is being sampled).

To confirm, then we could essentially do the same with this `planner->SendCommand("DumpTree")`?

Reply:
I wonder if `_configurations` in `ConstraintFilterReturn` also essentially provides the same information (i.e. robot configuration used during collision checking)